neuro_san/session/direct_agent_session.py

```diff
@@ -212,11 +212,12 @@ def streaming_chat(self, request_dict: Dict[str, Any]) -> Generator[Dict[str, An
                     response_dict["response"] = message
                     yield response_dict
         finally:
-            with contextlib.suppress(Exception):
-                # We are in a sync run-time environment,
-                # so we need to run the async delete_resources() method
-                # with the asyncio.run() helper.
-                asyncio.run(chat_session.delete_resources())
+            # Release resources without exceptions
+            with suppress(Exception):
+                # Cannot run as if in sync environment, so run async
+                # Use the asyncio_executor so as to not induce other async warnings
+                future: Future = asyncio_executor.submit(self.request_id, chat_session.delete_resources)
+                _ = future
```

Comment:
This is one of the real fixes:
Do not use asyncio.run() because doing that will cause a warning about how delete_resources() is not awaited.  Instead use the asyncio_executor which is at hand.